digital_land/api.py

```diff
@@ -0,0 +1,74 @@
+import csv
+import os
+import requests
+import logging
+
+
+class API:
+    def __init__(self, url, cache_dir=None):
+        self.url = url
+        self.cache_dir = cache_dir
+
+    def download_dataset(self, dataset):
+        entities = []
+        count = 0
+        url = f"{self.url}/entity.json?dataset={dataset}&limit=500"
+
+        logging.info(f"Downloading dataset {dataset}")
+
+        while url:
```

Comment:
If you use the files cdn as mentioned above you don't need to do this batching. Also the name of url is confusing here as it starts out as a url and then you convert it into a response